cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -172,6 +201,130 @@ func (s *Server) GetAvailablePackages(ctx context.Context, request *corev1.GetAv
 	}, nil
 }
 
+// GetPackageMeta streams the package metadata based on the request.
+func (s *Server) GetPackageMeta(ctx context.Context, request *corev1.GetPackageMetaRequest) (*corev1.GetPackageMetaResponse, error) {
+	log.Infof("+GetPackageMeta()")
+
+	url, err := s.pullChartTarball(ctx, request.Package)
+	if err != nil {
+		return nil, err
+	}
+	log.Infof("Found chart url: [%s]", *url)
+
+	// unzip and untar .tgz file
+	meta, err := fetchMetaFromChartTarball(request.Package.Name, *url)
+	if err != nil {
+		return nil, err
+	}
+
+	return &corev1.GetPackageMetaResponse{
+		Meta: &corev1.GetPackageMetaResponse_PackageMeta{
+			Readme: meta[readme],
+		},
+	}, nil
+}
+
+func (s *Server) pullChartTarball(ctx context.Context, packageRef *corev1.PackageReference) (*string, error) {
+	client, err := s.GetClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceIfc := client.Resource(chartsResource).Namespace("default")
```

Comment:
Why is the namespace hard-coded here? Fine if it's just for demoning right now.

Reply:
just a placeholder, like I said, a lot of refactoring to be done